docs/getting-started-guides/scratch.md

```diff
@@ -0,0 +1,664 @@
+process.  There are examples of Saltstack, Ansible, Juju, and CoreOS Cloud Config in the
+various Getting Started Guides.
+
+### Bootstrapping the Cluster
+
+While the basic node services (kubelet, kube-proxy, docker) are typically started and managed using
+traditional system administration/automation approaches, the remaining *master* components of Kubernetes are
+all configured and managed *by Kubernetes*:
+  - their options are specified in a Pod spec (yaml or json) rather than an /etc/init.d file or
+    systemd unit.
+  - they are kept running by Kubernetes rather than by init.
+  - they run in containers, and so tools like `ps` and `top` are less useful than `docker ps` and `kubectl get
+    pods`
+
+#### etcd
+You will need to run one or more instance of etcd.  
+  - Recommended approach: run one etcd instance, with its log written to a directory backed
+    by durable storage (RAID, GCE PD)
+  - Alternative: run 3 or 5 etcd instances.
+    - Log can be written to non-durable storage because storage is replicated. 
+    - run a single apiserver which connects to one of the etc nodes.
+ See [Availability](../availability.md) for more discussion on factors affecting cluster
+availability.
+
+To run an etcd instance:
+  - copy `cluster/saltbase/salt/etcd/etcd.manifest`
+  - make any modifications needed
+  - start the pod by putting it into the kubelet manifest directory
+
+#### Apiserver
+To run the apiserver:
+  - select the correct flags for your cluster
+  - write a pod spec for the apiserver using the provided template
+  - start the pod by putting it into the kubelet manifest directory
+
+Here are some apiserver flags you may need to set:
+  - `--cloud_provider=`
+  - `--cloud_config=` if cloud provider requires a config file (gce, aws). If so, need to put config file into apiserver docker image or mount through hostDir.
+  - `--address=${MASTER_IP}`.  
+   - or `--address=127.0.0.1` if you want to run a proxy on the master node.
+  - `--cluster_name=$CLUSTER_NAME`
+  - `--service-cluster-ip-range=$SERVICE_CLUSTER_IP_RANGE`
+  - `--etcd_servers=http://127.0.0.1:4001`
+  - `--tls_cert_file=/srv/kubernetes/server.cert` -%}
+  - `--tls_private_key_file=/srv/kubernetes/server.key` -%}
+  - `--admission_control=$RECOMMENDED_LIST`
+    - See [admission controllers](../admission_controllers.md) for recommended arguments.
+  - `--allow_privileged=true`, only if you trust your cluster user to run pods as root.
+ 
+If you are following the firewall-only security approach, then use these arguments:
+  - `--token_auth_file=/dev/null`
+  - `--insecure-bind-address=$MASTER_IP`
+  - `--advertise-address=$MASTER_IP`
+
+If you are using the HTTPS approach, then set:
+  - `--client_ca_file=/srv/kubernetes/ca.crt`
+  - `--token_auth_file=/srv/kubernetes/known_tokens.csv`
+  - `--basic_auth_file=/srv/kubernetes/basic_auth.csv`
+
+*TODO* document proxy-ssh setup.
+
+##### Apiserver pod template
```

Comment:
Move this whole section about the template under the "write a pod spec for the apiserver using the provided template" (399), so these options show under the corresponding step in the process.